### PR TITLE
fix: Duplicated options in Select when using numerical values

### LINKED ISSUE #24877: Numeric filters show duplicate values and UI issue when scrolling
In numeric chart filters and dashboard filters, after typing and selecting a value in the filter, duplicate values start to show up. 

#### How to reproduce the bug

1. Go to a chart 
2. Add a filter using a numeric column 
3. Type a value into the filter, then select the filter value 
4. Type another value into the filter and select the filter value 

### Expected results

One of each value with the selected ones highlighted 

### Actual results

Duplicate values start to appear. 
After scrolling up and down, more duplicate values show up and replace the actual filter values

#### Screenshots

https://github.com/apache/superset/assets/10627051/ee13139f-0723-49e5-9e25-2fc55035caff

### Environment

(please complete the following information):

- browser type and version: Chrome
- superset version: `master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
We noticed that `Select` options were being duplicated when searching for a value in numerical filters. It turns out that Ant Design Select can't have options with numerical values when mode equals `tags`. It displays the following warning:

`Warning: value of Option should not use number type when mode is tags or combobox`

To overcome this limitation, this PR changes the mode to `multiple` and implement missing features such as allowing new values and copy and paste.

Fixes https://github.com/apache/superset/issues/24877

### TESTING INSTRUCTIONS
I added a test to the component but we need to check the general use of selects in the application because there are wrappers that override the component behavior such as `SelectControl` and `SelectFilterPlugin`.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@michael-s-molina Have you considered an alternative approach of stringifying the numeric values when using them as option values and then parsing the numbers in onSelect callback? If it's possible, it could save us some added complexity of the features added for parity with tag mode

File: superset-frontend/src/components/Select/AsyncSelect.test.tsx [hunk 1]
```diff
@@ -323,12 +333,14 @@ test('same case should be ranked to the top', async () => {
   }));
   render(<AsyncSelect {...defaultProps} options={loadOptions} />);
   await type('Ac');
-  const options = await findAllSelectOptions();
-  expect(options.length).toBe(4);
-  expect(options[0]?.textContent).toEqual('acbc');
-  expect(options[1]?.textContent).toEqual('CAc');
-  expect(options[2]?.textContent).toEqual('abac');
-  expect(options[3]?.textContent).toEqual('Cac');
+  await waitFor(() => {
```
Comment (kgabryje) [inline superset-frontend/src/components/Select/AsyncSelect.test.tsx:336]: Are all those test changes relevant to the fix or are those just some bycatches?

**Comment by michael-s-molina** [comment]
> @michael-s-molina Have you considered an alternative approach of stringifying the numeric values when using them as option values and then parsing the numbers in onSelect callback? If it's possible, it could save us some added complexity of the features added for parity with tag mode

I did consider this approach but rejected for the following reasons:
- It goes against the Typescript definition of the values
- It has performance implications when dealing with a big number of options
- Numbers are supported in single and multiple mode.

File: superset-frontend/src/components/Select/AsyncSelect.test.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/Select/AsyncSelect.test.tsx:336]: These changes are the result of debouncing `handleOnSearch`.

**Comment by kgabryje** [comment]
> I did consider this approach but rejected for the following reasons:
> 
> * It goes against the Typescript definition of the values
> * It has performance implications when dealing with a big number of options
> * Numbers are supported in `single` and `multiple` modes.

Got it, thanks for explanation 👍

**Review by kgabryje — APPROVED** [review]
LGTM

**Review by geido — APPROVED** [review]
Code LGTM. We will run regression testing on this one to catch any possible side effect. Thanks!

**Pull request merged**